### ISSUE TITLE
Closing the command prompt is not handled on Windows

### ISSUE THREAD

**Issue by skulblakka** [streamlink/streamlink#2550]
## Bug Report

- [x ] This is a bug report and I have read the contribution guidelines.

### Description

When the user closes the command prompt (cmd/PowerShell) using the "X" button it is not handled by streamlink. This causes Windows to just kill it without giving streamlink the chance to terminate the player.

### Expected / Actual behavior

The player should be closed along streamlink when the command prompt window is closed.

### Reproduction steps / Explicit stream URLs to test

Using ```--player-passthrough``` or ```--player-args="--keep-open=yes {filename}"``` with mpv will keep the player open when closing the command prompt window. 

Additionally using CTRL-C does not work when using ```--player-passthrough```. Nothing happens until you close the player yourself or the stream has ended.

### Log output

I don't think log output is very useful here. If you disagree just tell me and I add some :)

### Additional comments, screenshots, etc.

I tried fixing this by [setting a Console Ctrl Handler](https://docs.microsoft.com/en-us/windows/console/setconsolectrlhandler) using win32api ([pypiwin32](https://pypi.org/project/pypiwin32)). 

I ran into an issue when using ```--player-passthrough```. The player is started using ```subprocess.call``` instead of ```subprocess.Popen```. As subprocess.call does not return until the subprocess is finished we can not kill the process when the handler is triggered*. I "fixed" this by not using call and waiting in the main thread instead. This also resolved the problem with streamlink not reacting to CTRL-C.

I also tested this on debian. I does not seem to break anything.

**I did not make a PR for this as I'm sure there is some room for improvement. I'm also not sure how to add the pypiwin32 dependency only for Windows.** My current solution is [here](https://gist.githubusercontent.com/skulblakka/7adacdd155b3bafedef31969cf932eee/raw/af545a222edcf6ff389c4b17fef4846d907fd5d9/win32_exit.diff). Hopefully someone can take a look at it.

_*This is just my observation on Windows. The same stuff works on Linux so I guess this is either Windows specific or my limited Know-How regarding python/this project._

My local setup:
- Windows 10 1809
- Python 3.7.3 x64
- streamlink commit 0f01f35

[Love Streamlink? Please consider supporting our collective. Thanks!](https://opencollective.com/streamlink/donate)

**Comment by beardypig** [comment]
Thanks the detailed issue report. It looks like `call` has been used since `--player-passthrough` was added, the reason (I assume) it because in Linux and macOS the player process would end if Streamlink were closed, this doesn't appear to be the case in Windows. The current setup was probably the easiest, least worst way to do it. I'd say you should PR your changes so we can better discuss it, it's looks pretty good - except the busy loop, which I'd change to check the status of the subprocess and return from the function when it exits. 
As for adding a dependency to for Windows only, that is relatively simple and doesn't cause too much issue for packaging (I think all the Windows packages are based off the install we build, so there is less of a barrier to add something). Obviously other people will want to review the changes, and we'll want to do some more testing on Linux/macOS :)

**Comment by gravyboat** [comment]
I've commented [on the other issue](https://github.com/streamlink/streamlink/issues/2535#issuecomment-511227705) about this.

**Comment by skulblakka** [comment]
I opened the PR before I saw your comment @gravyboat. I'm of course open for discussion whether or not this is or something similar actually needed/wanted behavior. 

This does however make streamlink on Windows behave like on Linux and fixes the issue that it can not be terminated using CTRL-C when using ```player-passthrough```.

**Comment by gravyboat** [comment]
@skulblakka Yeah my comment was mostly aimed at @beardypig since this behavior seems normal to me.
